addons/mail/static/src/model/model_manager.js

```diff
@@ -454,14 +448,17 @@ export class ModelManager {
                         throw new Error(`Relational field "${model}/${fieldName}" has "sort" with a relation of type "${field.relationType}" but "sort" is only supported for "many".`);
                     }
                 }
-                // 3. Check for redundant attributes on identifying fields.
+                // 3. Check for redundant or unsupported attributes on identifying fields.
                 if (field.identifying) {
                     if ('readonly' in field) {
                         throw new Error(`Identifying field ${model}/${fieldName} has unnecessary "readonly" attribute (readonly is implicit for identifying fields).`);
                     }
                     if ('required' in field && model.identifyingMode === 'and') {
                         throw new Error(`Identifying field ${model}/${fieldName} has unnecessary "required" attribute (required is implicit for AND identifying fields).`);
                     }
+                    if ('default' in field) {
+                        throw new Error(`Identifying field ${model}/${fieldName} has "default" attribute, but default values are not supported for identifying fields.`);
```

Comment:
Note that if someone wants this, could be that the record(s) that rely on default value can be modelled specifically.

For example, if we wanted `Thread` that has `model` with default `mail.channel`, just introduce `Channel`.